exchange.c

```diff
@@ -1,51 +1,76 @@
 #define _GNU_SOURCE
 
 #include <fcntl.h>
+#include <getopt.h>
+#include <gnu/libc-version.h>
+#include <linux/fs.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdbool.h>
+#include <sys/syscall.h>
+#include <unistd.h>
 
-void printHelp() {
-    puts("exchange: exchange names of two files or directories atomically\n"
-         "Usage: exchange FILE1 FILE2 [flags...]\n"
-         "Allowed flags:\n"
-         "\t--help:    shows this help text\n"
-         "\t--verbose: produces verbose output");
-}
+static struct option long_options[] = {
+    {"help",    no_argument, NULL, 'h'},
+    {"verbose", no_argument, NULL, 'v'},
+    {0, 0, 0, 0}
+};
 
-int main(int argc, char** argv) {
-    if (argc < 3) {
-        printHelp();
-        return 1;
-    }
+void usage(int exit_code) {
+    puts("Usage: exchange [-v] PATH1 PATH2\n"
+         "  or:  exchange -h|--help\n"
+         "atomically exchange names and contents of two files or directories\n"
```

Comment:
```suggestion
         "   or:  exchange -h|--help\n"
         "atomically exchange names of two files or directories\n"
```

According to `man renameat2`, the contents are not modified. Just the name is swapped.

Reply:
Fair enough. I find it a bit easier to reason about "names and contents" for what renameat2 does, but it's a very narrow point. I'll remove that.